connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -222,6 +227,21 @@ protected Producer<K, V> createProducer() {
             protected Consumer<K, V> createConsumer() {
                 return consumer;
             }
+
+            @Override
+            protected boolean readPartition(TopicPartition topicPartition) {
+                return readTopicPartition.test(topicPartition);
+            }
+
+            @Override
+            public void stop() {
+                super.stop();
+                // Close the clients here, if the thread that was responsible for closing them was never started.
+                if (thread == null) {
```

Comment:
IMO this check isn't necessary; `close` is idempotent for producers and consumers.